pkgs/applications/networking/cluster/nerdctl/default.nix

```
@@ -0,0 +1,34 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+let
+  commitSha = "da2ffa5cb5c9920a6d86a4c7466cce02780b774a";
+in
+buildGoModule rec {
+  pname = "nerdctl";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "AkihiroSuda";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "181qapqgp7zd0imk0zkn4wzpsw292ai2yz9pbiirpjcjx9h26w5h";
+  };
+
+  vendorSha256 = "0scywhllxk1m6456wggdmn7sgvy5x3gz2xnyfq9jnvvzap8byr2v";
+
+  buildFlagsArray = [
+    "-ldflags="
+    "-w"
+    "-s"
+    "-X github.com/AkihiroSuda/nerdctl/pkg/version.Version=v${version}"
+    "-X github.com/AkihiroSuda/nerdctl/pkg/version.Revision=${commitSha}"
```

A commit hash that hasn't been updated in a version bump often gets overlooked in reviews, the bot won't update it and if the revision is overridden it will be incorrect unless it is also updated to match. I'd suggest just using `src.rev`.

```suggestion
    "-X github.com/AkihiroSuda/nerdctl/pkg/version.Revision=${src.rev}"
```

---

I've tried it before but it's just the same value as `${version}`

https://github.com/NixOS/nixpkgs/issues/106543

---

Yes, that's correct. Currently there isn't an easy way to get the git hash without setting it manually which then gets overlooked.

---

:+1: I'm happy to leave it empty

---

`src.rev` is useful if the `revision` is overridden to point to another commit.

---

Ok. I'm now specifically setting it to it's default of `<unknown>`

https://github.com/06kellyjac/nixpkgs/blob/nerdctl/pkgs/applications/networking/cluster/nerdctl/default.nix#L35